components/server/src/server.rs

```diff
@@ -1874,9 +1880,34 @@ impl EnginesResourceInfo {
         if let Some(raft_engine) = &self.raft_engine {
             fetch_engine_cf(raft_engine, CF_DEFAULT, &mut normalized_pending_bytes);
         }
-        for cf in &[CF_DEFAULT, CF_WRITE, CF_LOCK] {
-            fetch_engine_cf(&self.kv_engine, cf, &mut normalized_pending_bytes);
+
+        let mut cached_latest_kv_pending_bytes =
+            self.cached_latest_kv_pending_bytes.as_ref().lock().unwrap();
+        // cached_latest_kv_pending_bytes records the compaction pending bytes of the
+        // tablet with the latest suffix for each region in each regular update. Clear
+        // ensures that detroyed tablet will not be in it forever.
+        cached_latest_kv_pending_bytes.clear();
+        self.tablet_factory
+            .for_each_opened_tablet(&mut |id, suffix, db: &RocksEngine| {
```

Comment:
The for_each_opened_tablet will acquire the lock, which blocks other operations such as split or apply a snapshot. 
So the key here is to figure out the total time of this function when we have 1000 instances, for example. If it takes seconds, then this approach will greatly impact P99 latency.

Reply:
Yes. Using for_each_opend_tablet to fetch pending bytes will occupy the lock for a long time. Here's a bench comparison between fetching pending bytes and just cloning them. The difference is so noticable.
![Kx2H1VCJoP](https://user-images.githubusercontent.com/71589810/183855220-687d6588-31a7-4b53-8d4f-2ff4e75f881e.png)
![rftpc1ZzDL](https://user-images.githubusercontent.com/71589810/183855238-30bb93a8-30f6-4a58-bf35-c7f9ab717178.png)